### ISSUE TITLE
False positive no-name-in-module with tensorflow==2.5.0

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#4609]
### Steps to reproduce

```bash
echo "from tensorflow.keras.applications import vgg16" | pylint --from-stdin bug | grep " E"
```

### Current behavior

```
bug:1:0: E0611: No name 'applications' in module 'LazyLoader' (no-name-in-module)
```

### Expected behavior

No error

### pylint --version output

Result of `pylint --version` output:

```
pylint 3.0.0a3
astroid 2.6.0
Python 3.9.5 (default, May  4 2021, 15:55:26) 
[GCC 7.5.0]
```

**Comment by DudeNr33** [comment]
What OS are you running?
I noticed similar errors with Python 3.9.5 on macOS.
Python 3.8 instead runs just fine.
So does Python 3.9.5 on Linux.

In my case those errors come from things like ``socket.error`` or ``math.sin``, which are both modules written in C.
``astroid`` has trouble with the inference in this case, but I do not know why yet.

**Comment by bersbersbers** [comment]
This is one OpenSUSE Leap 15.2.

**Comment by DudeNr33** [comment]
Can you try to reproduce this on OpenSUSE but with Python 3.8?

**Comment by bersbersbers** [comment]
> Can you try to reproduce this on OpenSUSE but with Python 3.8?

Same thing:

```bash
pyenv install 3.8.10
pyenv local 3.8.10
pip install tensorflow
pip install pylint astroid --pre -U
echo "from tensorflow.keras.applications import vgg16" | pylint --from-stdin --errors-only bug
pylint --version
```

gives

```text
************* Module bug
bug:1:0: E0611: No name 'applications' in module 'LazyLoader' (no-name-in-module)

pylint 3.0.0a3
astroid 2.6.0
Python 3.8.10 (default, Jan  1 1970, 00:00:00) 
[GCC 7.5.0]
```

**Comment by MajorCarrot** [comment]
We are having the same issue in the [pipeline](https://gitlab.com/adithyav1511/betsi/-/jobs/1497447685) of one of our projects
Is there any solution for this?

**Comment by dc5048** [comment]
[@DudeNr33  Wrote](https://github.com/PyCQA/pylint/issues/4609#issuecomment-869176670):
> I noticed similar errors with Python 3.9.5 on macOS.
> Python 3.8 instead runs just fine.
> So does Python 3.9.5 on Linux.
> 
> In my case those errors come from things like `socket.error` or `math.sin`, which are both modules written in C.

Since upgrading python to 3.9.6 on macOS this AM, I've been having the same issue with false positives when importing from the math module.

**Comment by DudeNr33** [comment]
@dc5048 I am not sure if this has the same reason (the problem with macOS and ``socket``/``math`` is also discussed here: #4759), but it would be interesting to know how you installed/upgraded your Python version. Did you install it directly from the website, or used homebrew, or something like ``pyenv``?
(maybe this discussion should be continued in #4759)

**Comment by dc5048** [comment]
@DudeNr33 , I used homebrew to update Python.  Thanks for the pointer to the other issue, I'll take a look.

**Comment by franneck94** [comment]
Same for me on Windows 10 with Anaconda Python 3.8.11 and Tensorflow 2.6.0
